packages/flutter_tools/lib/src/application_package.dart

```diff
@@ -81,13 +82,25 @@ class AndroidApk extends ApplicationPackage {
 }
 
 class IOSApp extends ApplicationPackage {
-  static const String _defaultId = 'io.flutter.runner.Runner';
-  static const String _defaultPath = 'ios/.generated';
-
   IOSApp({
-    String localPath: _defaultPath,
-    String id: _defaultId
-  }) : super(localPath: localPath, id: id);
+    String iosProjectDir,
+    String iosProjectBundleId
+  }) : super(localPath: iosProjectDir, id: iosProjectBundleId);
+
+  factory IOSApp.fromBuildConfiguration(BuildConfiguration config) {
+    if (getCurrentHostPlatform() != HostPlatform.mac) {
+      return null;
```

Comment:
From a factory, it should be legal. It's a tiny bit odd that factories look like constructors in Dart. So I am not sure if this is idiomatic.